runtime/config-deployer-service/ballerina/APIClient.bal

```diff
@@ -809,33 +821,53 @@ public class APIClient {
         if operationPoliciesToUse is APIOperationPolicies {
-            APKOperationPolicy[]? request = operationPoliciesToUse.request;
+            APKOperationPolicy[]? requestPolicies = operationPoliciesToUse.request;
+            APKOperationPolicy[]? responsePolicies = operationPoliciesToUse.response;
+            if requestPolicies is APKOperationPolicy[] && requestPolicies.length() > 0 {
+                model:HTTPRouteFilter headerModifierFilter = {'type: "RequestHeaderModifier"};
+                headerModifierFilter.requestHeaderModifier = self.extractHttpHeaderFilterData(requestPolicies, organization);
+                routeFilters.push(headerModifierFilter);
+            }
+            if responsePolicies is APKOperationPolicy[] && responsePolicies.length() > 0 {
+                model:HTTPRouteFilter headerModifierFilter = {'type: "ResponseHeaderModifier"};
+                headerModifierFilter.responseHeaderModifier = self.extractHttpHeaderFilterData(responsePolicies, organization);
+                routeFilters.push(headerModifierFilter);
+            }
         }
         return routeFilters;
     }
 
     isolated function extractHttpHeaderFilterData(APKOperationPolicy[] operationPolicy, commons:Organization organization) returns model:HTTPHeaderFilter {
+        model:HTTPHeader[] addPolicies = [];
         model:HTTPHeader[] setPolicies = [];
         string[] removePolicies = [];
         foreach APKOperationPolicy policy in operationPolicy {
-            string policyName = policy.policyName;
-
-            record {}? policyParameters = policy.parameters;
-            if (policyParameters is record {}) {
-                if (policyName == "addHeader") {
-                    model:HTTPHeader httpHeader = {
-                        name: <string>policyParameters.get("headerName"),
-                        value: <string>policyParameters.get("headerValue")
-                    };
-                    setPolicies.push(httpHeader);
+            if policy is HeaderModifierPolicy {
+                HeaderModifierPolicyParameters policyParameters = policy.parameters;
+                if policy.policyName == "AddHeaders" {
+                    ModifierHeader[] addHeaders = <ModifierHeader[]>policyParameters.headers;
+                    foreach ModifierHeader header in addHeaders {
+                        addPolicies.push(header);
+                    }
+                }
+                if policy.policyName == "SetHeaders" {
+                    ModifierHeader[] setHeaders = <ModifierHeader[]>policyParameters.headers;
+                    foreach ModifierHeader header in setHeaders {
+                        setPolicies.push(header);
+                    }
                 }
-                if (policyName == "removeHeader") {
-                    string httpHeader = <string>policyParameters.get("headerName");
-                    removePolicies.push(httpHeader);
+                if policy.policyName == "RemoveHeaders" {
```

Comment:
You can use `match` here - https://ballerina.io/learn/by-example/match-statement/

Reply:
Fixed in 562f33e530287161d01a5994da7c194a9fbabdc3

Reply:
Fixed in [2f908b5](https://github.com/wso2/apk/pull/2381/commits/2f908b5f0058cf3164e287d46e0082d11a19832e)